### PR TITLE
Fix bracket auto newline indentation, keyword highlighting, nested bracket pair highlighting, quote `auto_close`

### PR DESCRIPTION
**Pull request by spaceunifyfifty**
Resolves https://github.com/zed-extensions/zed-kotlin/issues/5, resolves https://github.com/zed-extensions/zed-kotlin/issues/6

### CODE REVIEW

**Review by osiewicz — COMMENTED** [review]
We could add brackets.scm for both `()` and `{}` while we're at it

File: extension.toml
Comment (maxdeviant) [inline extension.toml:3]: Let’s leave off the version bump in this PR.

We can bump the version when we go to publish.

Generally it’s better to separate functional changes from bumping the version.

Comment (maxdeviant) [reply extension.toml:3]: > But I can see that all three of them comprise one pull request; you're saying to split it into two separate pull requests then?

Correct.

That way we can merge your functional change but then make additional changes before a v0.0.3 release (if needed/desired).

File: languages/kotlin/highlights.scm
Comment (osiewicz) [inline languages/kotlin/highlights.scm:266]: I really hate to be that guy, but it looks like new keywords use spaces for indent, whereas everything else uses tabs.

**Review by osiewicz — APPROVED** [review]
LGTM, great job. :)

File: languages/kotlin/highlights.scm
Comment (osiewicz) [reply languages/kotlin/highlights.scm:266]: I honestly wouldn't worry too much about it. If I were you, I'd just `git commit --fixup sha_of_original_commit` and call it a day.

Comment (osiewicz) [reply languages/kotlin/highlights.scm:266]: We generally tend to avoid that conversation altogether and use a standardized formatter like rustfmt.

**Comment by osiewicz** [comment]
There's nothing holding up this PR from getting merged IMHO. As for merging main into this branch, I don't think you need to do this as your changes are disjoint from LSP changes made by cholwell.

File: languages/kotlin/highlights.scm
Comment (maxdeviant) [reply languages/kotlin/highlights.scm:266]: We use spaces, not tabs. Could you put the spaces back, please?

**Review by maxdeviant — CHANGES_REQUESTED** [review]
Can we standardize on spaces for indentation? It's what the rest of our languages do.

**Comment by osiewicz** [comment]
There's nothing from my side, no. I think @maxdeviant has merge rights.

**Review by maxdeviant — APPROVED** [review]

**Pull request merged**
